solc/CommandLineInterface.cpp

```diff
@@ -1191,8 +1192,11 @@ void CommandLineInterface::outputCompilationResults()
 	{
 		if (!m_options.output.dir.empty())
 			sout() << "Compiler run successful. Artifact(s) can be found in directory " << m_options.output.dir << "." << endl;
-		else
-			serr() << "Compiler run successful, no output requested." << endl;
+		else if (contracts.empty())
+			sout() << "Compiler run successful, no contracts to compile." << endl;
+		else if (contracts.size() > 0)
+			if (SemVerVersion{string(solidity::frontend::VersionString)}.isPrerelease())
+				sout() << "Compiler run successful. No output requested." << endl;
```

Comment:
Please add a test case covering this message too.

Reply:
@cameel Are there instructions anywhere on how to build a release version? This message doesn't output on pre-release versions.

Reply:
You need to create an empty `prerelease.txt` file at repo root. This is what our CI build script does when you set the `FORCE_RELEASE` variable: https://github.com/ethereum/solidity/blob/591df042115c6df190faa26a1fb87617f7772db3/scripts/ci/build.sh#L15

Normally `prerelease.txt` provides the extra bit that is included in the `solc --version` output of pre-release versions. When the file is there but is empty, the build config assumes you're building a release.

Reply:
@cameel Thank you for providing those instructions. So I have added a test to cover this situation however there are a number of cmdline tests that are failing that are in a format similar to this:

```
// SPDX-License-Identifier: GPL-3.0
pragma solidity >=0.0;
contract C {
	function f(uint a, uint b) public pure returns (uint, uint) {
		require(b != 0);
		return (a / b, a % b);
	}
}
```

They are only failing on the release build as the prerelease does not output the message stating there was nothing generated. Is there a way to only run these tests on the prerelease build?

Reply:
Oh. Looks like we have a problem. I see now that the SMTChecker does not output anything when it does not detect any problems.

In that case let's just modify `cmdlineTests.sh`. Please make it strip the no output message. The script already strips pre-release warning so it will be enough to make the no-output cases uniform for both release and pre-release builds.

So please modify this bit: https://github.com/ethereum/solidity/blob/591df042115c6df190faa26a1fb87617f7772db3/test/cmdlineTests.sh#L221-L227
to remove all three forms of the `Compiler run successful...` message, and only do it when the content of both `$stdout_expectation_file` and `$stderr_expectation_file` is empty.